### PR TITLE
#SA7F-7

### PR DESCRIPTION
**Pull request by Tubka**

### CODE REVIEW

File: src/pages/EditorTeam/EditorTeam.tsx [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+interface IPerson extends APIPerson{
+  lp?: number;
+  isEditMode?: boolean;
+}
+
+interface IProps {
+  row: IPerson
+  name: string;
+  onChange: any;
```
Comment (hendrysiak) [inline src/pages/EditorTeam/EditorTeam.tsx:60]: Missing type.

File: src/pages/EditorTeam/EditorTeam.tsx [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+}
+
+
+const CustomTableCell = ({ row, name, onChange }: IProps) => {
```
Comment (hendrysiak) [inline src/pages/EditorTeam/EditorTeam.tsx:64]: This file is over 300 lines length. Maybe we should split it into smaller files?

File: src/pages/EditorTeam/EditorTeam.tsx [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+        if (row.id === id) {
+          setPrevValue(row);
+          return { ...row, isEditMode: !row.isEditMode };
+        }
+        return row;
+      });
+    });
+  };
+
+  const onChange = (e: { target: { getAttribute: (arg0: string) => string; value: any; name: string; }; }, row: { id: string; }) => {
```
Comment (hendrysiak) [inline src/pages/EditorTeam/EditorTeam.tsx:147]: Very unreadable type. React has inside many types for events - use them!

File: src/pages/EditorTeam/EditorTeam.tsx [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+    if (e.target.getAttribute('name') === 'lp') return;
```
Comment (hendrysiak) [inline src/pages/EditorTeam/EditorTeam.tsx:148]: Maybe we should use enum?

File: src/pages/EditorTeam/EditorTeam.tsx [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+            surname: prevValue.surname
+          };
+        }
+        return row;
+      });
+      setRows(newRows);
+    }
+  };
+
+  const handleDelete = (id: string) => {
```
Comment (hendrysiak) [inline src/pages/EditorTeam/EditorTeam.tsx:188]: At this time:  
1\. Comment on this method (for future uses).  
2\. On delete we should save dateOfDelete - for counting fee.

In next prototype we can think about this case.

File: src/pages/EditorTeam/EditorTeam.tsx [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+          onClose={handleCloseNewMember}
+          aria-labelledby="simple-modal-title"
+          aria-describedby="simple-modal-description"
+        >
+          <NewTeamMember team={team}/>
+        </Modal>
+        
+      </div>
+      <Table className={classes.table} aria-label="caption table">
+        <caption>A barbone structure table example with a caption</caption>
```
Comment (hendrysiak) [inline src/pages/EditorTeam/EditorTeam.tsx:244]: Unused information?

File: src/pages/Team/components/TeamFinances/TeamFinances.tsx [hunk 7]
```diff
@@ -45,6 +45,7 @@ const TeamFinances = ({ incomes, currentTeam } : Props): JSX.Element => {
   useEffect(() => {
     const getData = async () => {
       const result = await axios.get(`/onePercent/${currentTeam}.json`);
+      console.log(result);
```
Comment (hendrysiak) [inline src/pages/Team/components/TeamFinances/TeamFinances.tsx:48]: Unused console.log.

File: src/store/reducers/income.ts [hunk 8]
```diff
@@ -148,15 +148,12 @@ const reducer = (state = initialState, action: ActionType): IncomeState => {
 
     case ActionTypes.DELETE_MEMBER:
       if (action.member.team) {
-        const teamAfterDelete = [...state.registry[action.member.team]];
-        const indexOfDeletedMember 
-          = teamAfterDelete.findIndex((m: APIPerson) => m.id === action.member.id);
-        
+        const teamAfterDelete = state.registry[action.member.team].filter(m => m.id !== action.member.id);
```
Comment (hendrysiak) [inline src/store/reducers/income.ts:151]: Very usefull change! Good idea!

**Review by hendrysiak — COMMENTED** [review]
I've added some comments to resolve. Happy coding!

File: src/App.css [hunk 9]
```diff
@@ -6,7 +6,7 @@
     text-align: center;
     overflow: hidden;
     position: relative;
-    background-color: #F4EEEB;
+    /* background-color: #F4EEEB; */
```
Comment (hendrysiak) [inline src/App.css:9]: Unused property

File: src/index.css [hunk 10]
```diff
@@ -40,7 +41,7 @@ code {
 }
 
 .Background {
-    background: linear-gradient(rgba(255, 255, 255, 0.5), rgba(255, 255, 255, 0.5)), url('./assets/Agnieszka_Madetko-Kurczab_18.jpg') center center/cover no-repeat;
+    /* background: linear-gradient(rgba(255, 255, 255, 0), rgba(255, 255, 255, 0)), url('./assets/Agnieszka_Madetko-Kurczab_18.jpg') center center/cover no-repeat; */
```
Comment (hendrysiak) [inline src/index.css:44]: Unused css

File: src/pages/EditorTeam/EditorTeam.tsx [hunk 11]
```diff
@@ -11,104 +8,44 @@ import DeleteIcon from '@material-ui/icons/Delete';
...
 const EditorTeam: FC = () => {
   const registry = useSelector((state: RootState) => state.income.registry);
   const [rows, setRows] = useState<IPerson[] | undefined>();
   const [team, setTeam] = useState<string>('');
-  const [previous, setPrevious] = useState<any>({});
   const classes = useStyles();
-  const [openNewMember, setOpenNewMember] = useState<boolean>(false);
   const [actualValue, setActualValue] = useState<IPerson>(
-    {name: '', surname:'', id: ''}
+    { name: '', surname:'', id: '', dateOfAdd: null }
```
Comment (hendrysiak) [inline src/pages/EditorTeam/EditorTeam.tsx:38]: And what about dateOfDelete?

Comment (Tubka) [reply src/pages/EditorTeam/EditorTeam.tsx:38]: Working great, type dateOfDelete is Date | null | undefined, są i don't have to declare this key in state.

**Pull request merged**